docs/DeveloperGuide.md

```diff
@@ -1,35 +1,298 @@
 # Developer Guide
 
+## Table of Contents
+
+* Table of Contents
+{:toc}
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+Adapted from AddressBook Level 3 Developer Guide (https://se-education.org/addressbook-level3/DeveloperGuide.html)
+
+## Setup
+
+### Setting up the project on your computer
+
+First, **fork** this repo, and **clone** the fork into your computer.
+
+If you plan to use Intellij IDEA (highly recommended):
+1. **Configure the JDK**: Follow the guide [_[se-edu/guides] IDEA: Configuring the JDK_](https://se-education.org/guides/tutorials/intellijJdk.html) to ensure Intellij is configured to use **JDK 11**.
+1. **Import the project as a Gradle project**: Follow the guide [_[se-edu/guides] IDEA: Importing a Gradle project_](https://se-education.org/guides/tutorials/intellijImportGradleProject.html) to import the project into IDEA.<br>
+   - Note: Importing a Gradle project is slightly different from importing a normal Java project.
+1. **Verify the setup**: Run `Main` and try a few commands.
+
+## Design
+
+> The `.puml` files used to create the diagrams in this guide can be found in the [diagrams](diagrams) folder.
+
+### Architecture
+
+![ArchitectureDiagram](images/ArchitectureDiagram.png)
+
+The above diagram provides a high-level overview of the program's design, which consists of 5 main components.
+
+- [`ToDoListManager`](#manager-component): Initializes the other components and connects them together
+- [`UI`](#ui-component): Handles all input and output between the user and the program
+- [`Logic`](#logic-component): Parses user input to commands and executes them
+- [`TaskList`](#tasklist-component): Holds the program's data in memory
+- [`Storage`](#storage-component): Saves data to the hard disk and loads it during the program's startup
+
+### Manager Component
+
+The code for this component is found in [`ToDoListManager.java`](../src/main/java/seedu/todolist/ToDoListManager.java).
+
+![ManagerClassDiagram](images/ManagerClassDiagram.png)
+
+The `ToDoListManager` component contains the `Ui`, `Logic`, `TaskList`, and `Storage` components, and depends
+on the `Command` class, which is returned by the `Parser` class and executed by the `ToDoListManager`.
+
+The `ToDoListManager` component,
+- initializes all the other components that it operates on (`Ui`, `Parser`, `TaskList`, `Storage`).
+- passes user input from the `Ui` component to the `Logic` component, which returns a `Command` object.
+- executes the returned `Command` object on the `TaskList` component, using the `Logic` component.
+- calls on the `Storage` component to save the data of the `TaskList` component to the hard disk.
+
+### Ui component
+
+The code for this component is found in [`Ui.java`](../src/main/java/seedu/todolist/ui/Ui.java).
```

Comment:
I think this link 404's? But great that you're using relative path.
Same for the following links below.